File: 프로그래머스/lv2/12981. 영어 끝말잇기/영어 끝말잇기.py
def solution(n, words):
    check = dict()
    check[words[0]] = True
    for i in range(1, len(words)):
        if check.get(words[i]) == None and words[i - 1][-1] == words[i][0]: 
            check[words[i]] = True
        else:
            return [(i % n) + 1, (i // n) + 1]
    return [0, 0]
        

File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.java
import java.util.*;
class Solution {
    static HashMap<String, Boolean> map = new HashMap<>();
    public int[] solution(int n, String[] words) {
        int idx = 1;
        int cnt = 1;
        int num = 0;
        for(String word : words){
            if(map.get(word) == null){
                map.put(word, true);
                idx ++;
            }
            else{
                return new int[]{idx, cnt};
            }
            
            if(num == 0){
                num += 1;
                continue;
            }
            if(words[num - 1].charAt(words[num - 1].length() - 1) != word.charAt(0)){
                return new int[]{idx - 1, cnt};
            }
            else{
                if(idx > n){
                    idx = 1;
                    cnt += 1;
                }
            }
            num += 1;
        }
        return new int[]{0, 0};
    }
}